module/VuFind/src/VuFind/Search/EDS/Results.php

```diff
@@ -131,4 +131,34 @@ public function getFacetList($filter = null)
         }
         return $this->buildFacetList($this->responseFacets, $filter);
     }
+
+    /**
+     * Get the scores of the results
+     *
+     * @return array
+     */
+    public function getScores()
+    {
+        $scoreMap = [];
+        foreach ($this->results as $record) {
+            $score = $record->getScore() ?? false;
+            if ($score) {
```

Comment:
I do not like this conditional -- I'd rather the list of scores be the same length as the list of results, and the record ID simply map to null or false if there is no score available.  But [Solr results](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/Search/Solr/Results.php#L165) does it this way.  Is it legit to change that implementation instead?